compiler/src/dmd/expressionsem.d

```diff
@@ -5272,8 +5272,12 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             }
 
             const(char)* failMessage;
-            if (!tf.callMatch(null, exp.argumentList, 0, &failMessage, sc))
+            ushort charIndex = 0;
+            if (!tf.callMatch(null, exp.argumentList, 0, &failMessage, sc, &charIndex))
             {
+                // set the charnum to the proper argument's position if any
+                if (charIndex > 0)
+                    exp.loc.charnum(charIndex);
```

Comment:
The location of the call expression shouldn't change, and the argument may be on another line than the function.

I suggest turning the `ushort* charIndex` into a `Loc* failLoc` and passing that to `.error` on line 5289.